### ISSUE TITLE
New pulp based scheduler is too verbose

### ISSUE THREAD

**Issue by ashwinvis** [snakemake/snakemake#603]
**Snakemake version** 5.23.0

**Describe the bug**
Importing pulp and using the pulp solver throws too many messages irrelevant to the user. This was not present in previous snakemake versions. This can be captured using one of these:
https://docs.python.org/3.8/library/contextlib.html?#contextlib.redirect_stdout
https://docs.python.org/3.8/library/warnings.html?#temporarily-suppressing-warnings

**Logs**
```sh
❯ snakemake -s test.smk hello -j                                                                                                                                                       
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 12
Rules claiming more threads will be scaled down.
Job counts:



/home/avmo/.pyenv/versions/3.8.5/envs/snek/lib/python3.8/site-packages/pulp/pulp.py:1195: UserWarning: Spaces are not permitted in the name. Converted to '_'
  warnings.warn("Spaces are not permitted in the name. Converted to '_'")
Welcome to the CBC MILP Solver 
Version: 2.9.0 
Build Date: Feb 12 2015 

command line - /home/avmo/.pyenv/versions/3.8.5/envs/snek/lib/python3.8/site-packages/pulp/apis/../solverdir/cbc/linux/64/cbc /tmp/71280860e1034f69855f4445ad637c66-pulp.mps max ratio None allow None threads None presolve on strong None gomory on knapsack on probing on branch printingOptions all solution /tmp/71280860e1034f69855f4445ad637c66-pulp.sol (default strategy 1)
At line 2 NAME          MODEL
At line 3 ROWS
At line 7 COLUMNS
At line 13 RHS
At line 16 BOUNDS
At line 18 ENDATA
Problem MODEL has 2 rows, 1 columns and 2 elements
Coin0008I MODEL read with 0 errors
String of None is illegal for double parameter ratioGap value remains 0
String of None is illegal for double parameter allowableGap value remains 0
String of None is illegal for integer parameter threads value remains 0
String of None is illegal for integer parameter strongBranching value remains 5
Option for gomoryCuts changed from ifmove to on
Option for knapsackCuts changed from ifmove to on
Continuous objective value is 2 - 0.00 seconds
Cgl0004I processed model has 0 rows, 0 columns (0 integer (0 of which binary)) and 0 elements
Cbc3007W No integer variables - nothing to do
Cuts at root node changed objective from -2 to -1.79769e+308
Probing was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Gomory was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Knapsack was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
Clique was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
MixedIntegerRounding2 was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
FlowCover was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)
TwoMirCuts was tried 0 times and created 0 cuts of which 0 were active after adding rounds of cuts (0.000 seconds)

Result - Optimal solution found

Objective value:                2.00000000
Enumerated nodes:               0
Total iterations:               0
Time (CPU seconds):             0.00
Time (Wallclock seconds):       0.00

Option for printingOptions changed from normal to all
Total time (CPU seconds):       0.00   (Wallclock seconds):       0.00


[Wed Sep  9 12:11:23 2020]
rule hello:
    jobid: 0

hello
[Wed Sep  9 12:11:23 2020]
Finished job 0.
1 of 1 steps (100%) done
Complete log: /home/avmo/src/snek5000/snek5000-abl/.snakemake/log/2020-09-09T121123.325408.snakemake.log
```
**Minimal example**
```snakemake
rule hello:
    shell: "echo hello"
```

**Additional context**
```sh
❯ pip show pulp           
Name: PuLP
Version: 2.3
Summary: PuLP is an LP modeler written in python. PuLP can generate MPS or LP files and call GLPK, COIN CLP/CBC, CPLEX, and GUROBI to solve linear problems.
Home-page: https://github.com/coin-or/pulp
Author: J.S. Roy and S.A. Mitchell
Author-email: pulp@stuartmitchell.com
License: UNKNOWN
Location: /home/avmo/.pyenv/versions/3.8.5/envs/snek/lib/python3.8/site-packages
Requires: amply
Required-by: snakemake

❯ python --version                                     
Python 3.8.5
```

**Comment by johanneskoester** [comment]
Thank you, fixed with commit ec538fc9c99054ad1b168d556a66f4a79aa41861. I will release a new version today.

**Comment by TBradley27** [comment]
Hello,

Apologies in advance, I also have a pulp related issue, and I thought it may be related to the issue posted here, so I wasn't sure whether to create a new issue or just post in this thread.

I am using snakemake version `5.24.0` in a clean environment with no packages installed other than mamba - and core snakemake processes seem to be broken I am afraid

**Installation**
```
conda create --name foo
conda activate foo
conda install -c conda-forge mamba
mamba install -c conda-forge -c bioconda snakemake
```

**Snakefile**
```
rule get_bar:
        output:
            "bar.txt"
        shell:
            "touch {output}"
```

**Error message:**
```
Traceback (most recent call last):
  File "/Users/bradle02/miniconda3/envs/foo/lib/python3.8/site-packages/snakemake/__init__.py", line 643, in snakemake
    success = workflow.execute(
  File "/Users/bradle02/miniconda3/envs/foo/lib/python3.8/site-packages/snakemake/workflow.py", line 961, in execute
    success = scheduler.schedule()
  File "/Users/bradle02/miniconda3/envs/foo/lib/python3.8/site-packages/snakemake/scheduler.py", line 406, in schedule
    else self.job_selector_ilp(needrun)
  File "/Users/bradle02/miniconda3/envs/foo/lib/python3.8/site-packages/snakemake/scheduler.py", line 615, in job_selector_ilp
    prob.solve()
  File "/Users/bradle02/miniconda3/envs/foo/lib/python3.8/site-packages/pulp/pulp.py", line 1890, in solve
    status = solver.actualSolve(self, **kwargs)
```
